crates/nu-command/src/network/url/parse.rs

```diff
@@ -0,0 +1,178 @@
+use super::url;
+use nu_protocol::ast::Call;
+use nu_protocol::engine::{Command, EngineState, Stack};
+use nu_protocol::{
+    Category, Example, PipelineData, ShellError, Signature, Span, SyntaxShape, Type, Value,
+};
+
+use url::Url;
+
+#[derive(Clone)]
+pub struct SubCommand;
+
+impl Command for SubCommand {
+    fn name(&self) -> &str {
+        "url parse"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("url parse")
+            .input_output_types(vec![(Type::String, Type::Record(vec![]))])
+            .rest(
+                "rest",
+                SyntaxShape::CellPath,
+                "optionally operate by cell path",
+            )
+            .category(Category::Network)
+    }
+
+    fn usage(&self) -> &str {
+        "Parses a url"
+    }
+
+    fn search_terms(&self) -> Vec<&str> {
+        vec!["scheme", "hostname", "port", "path", "parameters"]
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        _: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        parse(input.into_value(call.head), engine_state)
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![Example {
+            description: "Parses a url",
+            example: "'http://www.example.com:8081/foo/bar?p=section&s=&f[name]=vldc' | url parse",
+            result: Some(Value::Record {
+                cols: vec![
+                    "scheme".to_string(),
+                    "host".to_string(),
+                    "port".to_string(),
+                    "path".to_string(),
+                    "query".to_string(),
+                    "params".to_string(),
```

Comment:
Any interest in adding the other parsed components while you're at it?  I think username and password are pretty useful for people working with ssh, and fragment is common... may as well take them all?